.travis.yml

```diff
@@ -1,35 +1,28 @@
 sudo: false
 language: ruby
 cache: bundler
-before_install:
-   bundle -v
+before_install: bundle -v
 script: env COVERAGE=yes bundle exec rake
 matrix:
   fast_finish: true
   include:
-  - rvm: '2.5'
-    env: PUPPET_GEM_VERSION='~> 6.0'
-  - rvm: '2.4'
-    env: PUPPET_GEM_VERSION='~> 6.0'
-  - rvm: '2.3'
-    env: PUPPET_GEM_VERSION='~> 6.0'
-  - rvm: '2.4'
-    env: PUPPET_GEM_VERSION='~> 5.0'
-  - rvm: '2.3'
-    env: PUPPET_GEM_VERSION='~> 5.0'
-  - rvm: '2.1'
-    env: PUPPET_GEM_VERSION='~> 5.0'
-  - rvm: '2.4'
-    env: PUPPET_GEM_VERSION='~> 4.0'
-  - rvm: '2.3'
-    env: PUPPET_GEM_VERSION='~> 4.0'
-  - rvm: '2.1'
-    env: PUPPET_GEM_VERSION='~> 4.0'
-  - rvm: '2.1'
-    env: PUPPET_GEM_VERSION='~> 3.0'
-  - rvm: '2.0'
-    env: PUPPET_GEM_VERSION='~> 3.0'
-  - rvm: '1.9'
-    env: PUPPET_GEM_VERSION='~> 3.0'
+    - rvm: "2.5"
+      env: PUPPET_GEM_VERSION='~> 6.0'
+    - rvm: "2.4"
+      env: PUPPET_GEM_VERSION='~> 6.0'
+    - rvm: "2.3"
+      env: PUPPET_GEM_VERSION='~> 6.0'
+    - rvm: "2.4"
+      env: PUPPET_GEM_VERSION='~> 5.0'
+    - rvm: "2.3"
+      env: PUPPET_GEM_VERSION='~> 5.0'
+    - rvm: "2.1"
```

Comment:
This is interesting.

Puppet Agent 5.x (AIO) ships with Ruby 2.4. So I don't _think_ we need to test on 2.1?

That said, puppet gemspec for 5.x says it requires ruby 1.9.3+ .  The only reason I can think of for this is that puppet running on Puppet Server uses the old jruby which is 1.9.3 language compliant.

@rodjek  thoughts?

Ref: https://github.com/puppetlabs/puppet/blob/5.5.x/.gemspec#L21

Reply:
We've been bitten previously with modules using "modern" ruby syntax and then it failing in Puppet Server because it's running old jruby (1.9.x syntax)

IIRC we've updated to jruby 9k but I can't remember exactly when

HOWEVER...

that's puppet server runtime, not necessarily puppet spec testing time which is where spec_helper resides

---

So perhaps just remove Ruby 2.1 for Puppet 5.x and this is a 👍 from me!

Reply:
JRuby9k was added as optional in puppet-server 5.x and not switched over as the default until puppet-server 6.x, so we should still keep testing Puppet 4.x and 5.x against Ruby 1.9.3.